src/agent/communicator/include/ihttp_client.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <ihttp_socket.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/beast.hpp>
+
+#include <functional>
+#include <optional>
+#include <string>
+
+namespace http_client
+{
+    struct HttpRequestParams
+    {
+        boost::beast::http::verb Method;
+        std::string Host;
+        std::string Port;
+        std::string Endpoint;
+        std::string Token;
+        std::string User_pass;
+        std::string Body;
+
+        HttpRequestParams(boost::beast::http::verb method,
+                          const std::string& host,
+                          const std::string& port,
+                          const std::string& endpoint,
+                          const std::string& token = "",
+                          const std::string& user_pass = "",
+                          const std::string& body = "")
+            : Method(method)
+            , Host(host)
+            , Port(port)
+            , Endpoint(endpoint)
+            , Token(token)
+            , User_pass(user_pass)
+            , Body(body)
+        {
+        }
+
+        bool operator==(const HttpRequestParams& other) const
+        {
+            return Method == other.Method && Host == other.Host && Port == other.Port && Endpoint == other.Endpoint &&
+                   Token == other.Token && User_pass == other.User_pass && Body == other.Body;
+        }
+    };
+
+    class IHttpClient
+    {
+    public:
+        virtual ~IHttpClient() = default;
+
+        virtual boost::beast::http::request<boost::beast::http::string_body>
+        CreateHttpRequest(const HttpRequestParams& params) = 0;
```

Comment:
I think this function can be private.

Reply:
We can make it private, though we would have to remove tests that were written for it.